src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -110,6 +110,12 @@ def load_arguments(self, _):
         c.argument('disable_sgxquotehelper', action='store_true')
         c.argument('kubelet_config', type=str)
         c.argument('linux_os_config', type=str)
+        c.argument('appgw_name', options_list=['--appgw-name'], arg_group='Application Gateway')
+        c.argument('appgw_subnet_prefix', options_list=['--appgw-subnet-prefix'], arg_group='Application Gateway', deprecate_info=c.deprecate(redirect='--appgw-subnet-cidr', hide=True))
+        c.argument('appgw_subnet_cidr', options_list=['--appgw-subnet-cidr'], arg_group='Application Gateway')
+        c.argument('appgw_id', options_list=['--appgw-id'], arg_group='Application Gateway')
+        c.argument('appgw_subnet_id', options_list=['--appgw-subnet-id'], arg_group='Application Gateway')
+        c.argument('appgw_watch_namespace', options_list=['--appgw-watch-namespace'], arg_group='Application Gateway')
```

Comment:
Just a kind reminder, CLI core will automatically convert `_` into `-` in argument name for the long option. If you don't add a short option or don't change the long option name, you can omit the `options_list` part for future PRs.